packages/belos/tpetra/example/PCPG/PCPGTpetraExFile.cpp

```diff
@@ -0,0 +1,437 @@
+                   maxDeflate);  // Number of vectors in seed space
+    belosList->set("Num Saved Blocks",
+                   maxSave);                     // Number of vectors saved from old spaces
+    belosList->set("Orthogonalization", ortho);  // Orthogonalization type
+
+    if (numRhs > 1) {
+      belosList->set("Show Maximum Residual Norm Only",
+                     true);  // although numRhs = 1.
+    }
+    if (verbose) {
+      belosList->set("Verbosity", Belos::Errors + Belos::Warnings + Belos::TimingDetails +
+                                      Belos::FinalSummary + Belos::StatusTestDetails);
+      if (frequency > 0)
+        belosList->set("Output Frequency", frequency);
+    } else
+      belosList->set("Verbosity", Belos::Errors + Belos::Warnings + Belos::FinalSummary);
+    //
+    // *******Construct a preconditioned linear problem********
+    //
+    RCP<LinearProblem> problem = rcp(new LinearProblem(A, LHS, RHS));
+
+    // problem->setLeftPrec( prec );
+
+    bool set = problem->setProblem();
+    if (set == false) {
+      if (procVerbose)
+        std::cout << std::endl
+                  << "ERROR:  Belos::LinearProblem failed to set up correctly!" << std::endl;
+      return -1;
+    }
+
+    // Create an iterative solver manager.
+    RCP<PCPGSolMgr> solver = rcp(new PCPGSolMgr(problem, belosList));
+
+    //
+    // *******************************************************************
+    // ************************* Iterate PCPG ****************************
+    // *******************************************************************
+    //
+    if (procVerbose) {
+      std::cout << std::endl << std::endl;
+      std::cout << "Dimension of matrix: " << numGlobalElements << std::endl;
+      std::cout << "Number of right-hand sides: " << numRhs << std::endl;
+      std::cout << "Block size used by solver: " << blocksize << std::endl;
+      std::cout << "Maximum number of iterations allowed: " << maxIters << std::endl;
+      std::cout << "Relative residual tolerance: " << tol << std::endl;
+      std::cout << std::endl;
+    }
+    bool badRes;
+    for (int timeStep = 0; timeStep < numTimeStep; timeStep++) {
+      if (timeStep) {
+        // old epetra B->multiply(false, *vecLHS, *vecRHS); // rhs_new :=
+        // B*lhs_old,
+        B->apply(*LHS, *RHS);  // rhs_new := B*lhs_old,
+        set = problem->setProblem(LHS, RHS);
+        if (set == false) {
+          if (procVerbose)
+            std::cout << std::endl
+                      << "ERROR:  Belos::LinearProblem failed to set up correctly!" << std::endl;
+          return -1;
+        }
+      }  // if timeStep
+      std::vector<ST> rhs_norm(numRhs);
+      MVT::MvNorm(*RHS, rhs_norm);
+      std::cout << "                  RHS norm is ... " << rhs_norm[0] << std::endl;
+      //
+      // Perform solve
+      //
+      Belos::ReturnType ret = solver->solve();
+      //
+      // Compute actual residuals.
+      //
```

Comment:
Comments are sometimes:
```
//
// [...]
//
```
and sometimes `// [...]`.
Try to be consistent. I personally think that `//` is ok if it's a short comment like here